dashboard/app/models/curriculum_reference.rb

```diff
@@ -0,0 +1,42 @@
+# == Schema Information
+#
+# Table name: levels
+#
+#  id                       :integer          not null, primary key
+#  game_id                  :integer
+#  name                     :string(255)      not null
+#  created_at               :datetime
+#  updated_at               :datetime
+#  level_num                :string(255)
+#  ideal_level_source_id    :integer
+#  solution_level_source_id :integer
+#  user_id                  :integer
+#  properties               :text(65535)
+#  type                     :string(255)
+#  md5                      :string(255)
+#  published                :boolean          default(FALSE), not null
+#  notes                    :text(65535)
+#
+# Indexes
+#
+#  index_levels_on_game_id  (game_id)
+#  index_levels_on_name     (name)
+#
+
+class CurriculumReference < External
+  def dsl_default
+    <<-TEXT.strip_heredoc.chomp
+    # Update the following lines to give you level a unique name and point at
+    # the appropriate url on docs.code.org
+    name '#{name || 'unique level name here'}'
+    reference '/csd/maker_leds/index.html'
+    TEXT
+  end
+
+  # Get the URL of the studio.code.org/docs routes (that serves as a proxy to
+  # our docs.code.org route)
```

Comment:
Basically untrusted, not fully untrusted ;-)

My goal was not to protect from malicious content inside these iframes, but from accidental "bad" usage. Though escaping the iframe is very doable, it's not something that I expect will be done unintentionally.

My original approach had been to just direct this html directly into the parent document. In this case, it was trivial to have CSS from the child unintentionally pollute the parent